Source/BusinessLogic/Jobs/BoardGameGeekCleanUpService/BoardGameGeekCleanUpService.cs

```diff
@@ -0,0 +1,168 @@
+                                else
+                                {
+                                    var newRecord = CreateBGGGame(gameToAdd);
+
+                                    UpdateGameDefinition(game, newRecord.Id, result);
+                                }
+                            }
+                        }
+                    }
+
+                    if (game.BoardGameGeekGameDefinitionId != null)
+                    {
+                        _dataContext.CommitAllChanges();
+                    }
+                    else
+                    {
+                        result.StillOrphanGames.Add(new LinkOrphanGamesResult.OrphanGame()
+                        {
+                            Name = game.Name,
+                            Id = game.Id,
+                            GamingGroupId = game.GamingGroupId
+                        });
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                result.Success = false;
+                _rollbar.SendException(ex);
+            }
+
+            stopwatch.Stop();
+            result.TimeEllapsed = stopwatch.Elapsed;
+            return result;
+        }
+
+        private void UpdateGameDefinition(GameDefinition game, int boardGameGeekGameDefinitionId,
+            LinkOrphanGamesResult result)
+        {
+            game.BoardGameGeekGameDefinitionId = boardGameGeekGameDefinitionId;
+            result.LinkedGames++;
+        }
+
+        private BoardGameGeekGameDefinition CreateBGGGame(GameDetails gameToAdd)
+        {
+            var newRecord = new BoardGameGeekGameDefinition
+            {
+                Id = gameToAdd.GameId,
+                Name = gameToAdd.Name,
+                Thumbnail = gameToAdd.Thumbnail,
+                MaxPlayers = gameToAdd.MaxPlayers,
+                MinPlayers = gameToAdd.MinPlayers,
+                PlayingTime = gameToAdd.PlayingTime,
+                AverageWeight = gameToAdd.AverageWeight,
+                Description = gameToAdd.Description
+            };
+
+            _dataContext.Save(newRecord, new ApplicationUser());
+            return newRecord;
+        }
+
+        private BoardGameGeekGameDefinition GetExistingBGGGameById(GameDetails gameToAdd)
+        {
+            var existingGame = _dataContext.GetQueryable<BoardGameGeekGameDefinition>()
+                .FirstOrDefault(bgg => bgg.Id == gameToAdd.GameId);
+            return existingGame;
+        }
+
+        private GameDetails GetGameToAddFromSearch(List<SearchBoardGameResult> searchResult)
+        {
+            var gamesToAdd =
+                searchResult.Select(
+                    searchBoardGameResult => _boardGameGeekApiClient.GetGameDetails(searchBoardGameResult.BoardGameId))
+                    .Where(gameDetails => gameDetails != null)
+                    .ToList();
+
+            var gameToAdd =
+                gamesToAdd.Where(g => g.Thumbnail != null)
```

Comment:
Yes, you're right.
This "trick" is for select always one game if there are multiple exact results
Example: 
- https://www.boardgamegeek.com/xmlapi2/search?query=eclipse&type=boardgame&exact=1ç
- https://boardgamegeek.com/geeksearch.php?action=search&advsearch=1&objecttype=boardgame&q=eclipse&include%5Bdesignerid%5D=&exact=1&geekitemname=&include%5Bpublisherid%5D=&geekitemname=&range%5Byearpublished%5D%5Bmin%5D=&range%5Byearpublished%5D%5Bmax%5D=&range%5Bminage%5D%5Bmax%5D=&floatrange%5Bavgrating%5D%5Bmin%5D=&floatrange%5Bavgrating%5D%5Bmax%5D=&range%5Bnumvoters%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmin%5D=&floatrange%5Bavgweight%5D%5Bmax%5D=&range%5Bnumweights%5D%5Bmin%5D=&colfiltertype=&searchuser=kairos&nosubtypes%5B%5D=boardgameexpansion&range%5Bminplayers%5D%5Bmax%5D=&range%5Bmaxplayers%5D%5Bmin%5D=&playerrangetype=normal&range%5Bleastplaytime%5D%5Bmin%5D=&range%5Bplaytime%5D%5Bmax%5D=&B1=Submit

For "Eclipse" i want the most "famous" because i thing it's more possible to be the good one instead the others.
Anyway, i can improve this query trying to extract the year from the game name (if has) and filter with that data.
If there are not year in the name or we have several exact result we can leave it orphan if you want or apply some arbitrary rule than the BGG order...